hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/ethereum/EthTxData.java

```diff
@@ -45,6 +46,7 @@ public record EthTxData(
         BigInteger value, // weibar, always positive - note that high-bit might be ON in RLP encoding: still positive
         byte[] callData,
         byte[] accessList,
+        RLPList accessListAsRLP,
```

Comment:
This is awkward.  This is a fairly widely used class from a module that is intended to hide certain implementation details.  One of those details would be the use of the headlong library - from which the `RLPList` type comes.  And sure enough, the headlong library isn't exported from the module's `module-info.java`.

Problem is - what to replace it with?

No _good_ suggestions, but what I have is:  Since the only other use is in `EthTxSigs` in this very same module (and I wouldn't anticipate this field being used anywhere else in the near future): You _could_ hold it in the record as an `Object` - and then `EthTxSigs` would have to know the intimate details of `EthTxData` - but it _is_ in the same package ...

Reply:
Ok, I have a question, do you wish to get rid off using of headlong library in the `EthTxData` class or do not use RLPList as a field in this class? Because if you only wish to not use `RLPList` as an record than maybe the best solution would be holding this record field `accessListAsRLP` as an array of Object? Instead of using the new method in EthTxSigs
``` java
static Object[] encodeRLPList(RLPList rlpList) {
        return rlpList.elements().stream().map(rlpItem -> {
            if (rlpItem.isList()) {
                return encodeRLPList(rlpItem.asRLPList());
            }
            else {
                return rlpItem.data();
            }
        }).toArray();
    }
```
I could always move it to EthTxData and do the transformation here and save the result as the new record as an array of Object and then only use the result in calculating the signature. Would that be ok?

Reply:
RIght, not get rid of the use of headlong, just don't expose any headlong type (`RLPList` in this case) as a record field which means other modules will see it.

Holding it as an `Object[]` is just fine - just do whatever you need to do in `EthTxSigs`. Sound OK?